metals/src/main/scala/scala/meta/metals/search/InMemorySymbolIndex.scala

```diff
@@ -174,39 +176,45 @@ class InMemorySymbolIndex(
    *                 - filename must be formatted as a URI
    *                 - names must be sorted
    */
-  def indexDocument(document: s.Document): Effects.IndexSemanticdb = {
-    val uri = Uri(document.filename)
-    val input = Input.VirtualFile(document.filename, document.contents)
+  def indexDocument(document: TextDocument): Effects.IndexSemanticdb = {
+    val uri = Uri(document.uri)
+    val input = Input.VirtualFile(document.uri, document.text)
     documentIndex.putDocument(uri, document)
-    document.names.foreach {
+    document.occurrences.foreach {
       // TODO(olafur) handle local symbols on the fly from a `Document` in go-to-definition
       // local symbols don't need to be indexed globally, by skipping them we should
       // def isLocalSymbol(sym: String): Boolean =
       // !sym.endsWith(".") &&
       //     !sym.endsWith("#") &&
       //     !sym.endsWith(")")
       // be able to minimize the size of the global index significantly.
-      //      case s.ResolvedName(_, sym, _) if isLocalSymbol(sym) => // Do nothing, local symbol.
-      case s.ResolvedName(Some(s.Position(start, end)), sym, true) =>
+      //      case s.SymbolOccurrence(_, sym, _) if isLocalSymbol(sym) => // Do nothing, local symbol.
+      case SymbolOccurrence(Some(r), sym, SymbolOccurrence.Role.DEFINITION) =>
         symbolIndexer.addDefinition(
           sym,
-          i.Position(document.filename, Some(input.toIndexRange(start, end)))
+          i.Position(document.uri, Some(input.toIndexRange(r)))
         )
-      case s.ResolvedName(Some(s.Position(start, end)), sym, false) =>
+      case SymbolOccurrence(Some(r), sym, SymbolOccurrence.Role.REFERENCE) =>
         symbolIndexer.addReference(
-          document.filename,
-          input.toIndexRange(start, end),
+          document.uri,
+          input.toIndexRange(r),
           sym
         )
       case _ =>
     }
     document.symbols.foreach {
-      case s.ResolvedSymbol(sym, Some(denot)) =>
+      case denot: semanticdb3.SymbolInformation =>
```

Comment:
s/denot/symbolInfo ?

Reply:
Or simply `info`. This is how we name related variables in Scalameta.